kuryr_kubernetes/cni/daemon/service.py

```diff
@@ -359,13 +360,15 @@ def __init__(self, worker_id, metrics):
 
     def _start_metric_updater(self):
         while self.is_running:
-            if self.metrics:
-                pod_name = list(self.metrics.keys())[0]
-                with lockutils.lock(pod_name):
-                    labels = self.metrics[pod_name]['labels']
-                    duration = self.metrics[pod_name]['duration']
-                    self.prometheus_exporter.update_metric(labels, duration)
-                    del self.metrics[pod_name]
+            try:
+                metric = self.metrics.get(timeout=1)
```

Comment:
does the get here blocks the addition of new metrics while removing from queue?

Reply:
Not sure if I understand. This queue module should be thread safe, meaning that `get()` blocks until there's an item added using `put()`. Or the timeout happens, which we use to check `self.is_running` to make sure the thread is able to correctly stop even if there are no new CNI requests handled at a moment.

Reply:
excellent. Thanks for explaining.